src/hotspot/share/opto/superword.cpp

```diff
@@ -962,7 +966,7 @@ bool SuperWord::stmts_can_pack(Node* s1, Node* s2, int align) {
     return false; // No vectors for this type
   }
 
-  if (isomorphic(s1, s2)) {
+  if (isomorphic(s1, s2) && !is_populate_index(s1, s2)) {
```

Comment:
Note: As explained in `test7a`:
If we pack the `iv + 1`, `iv + 2`, `iv + 3` ... `AddI` nodes, then we get a pack that is non-power-of-2 in most cases, since `iv + 0` has collapsed to `iv`, and is not an `AddI` but a `Phi` node. That then causes `split_packs_at_use_def_boundaries` to cut the use-packs, and things unravel from there.

So I had 2 options:
1. Create `Phi, AddI, AddI, AddI, ...` packs, and deal with the consequences of having nodes of different types in a pack. Plus, we would have the `iv` in a pack, which also has consequences all over the place... Not great.
2. Forbid anything that looks like a `PopulateIndex` to be packed. It does not need to be packed, and will still be vectorized, see `SuperWord::vector_opd`. I went with this option.

Reply:
Agree with option 2.

Reply:
Great :)

Reply:
Yes, they did. It was some of those tests that motivated the introduction of `is_populate_index`

Reply:
Add comment  (your option 2.) before `!is_populate_index(` check to explain why we do check.

Reply:
Done